metagraph/src/annotation/binary_matrix/base/binary_matrix.hpp

```diff
@@ -32,9 +35,13 @@ class BinaryMatrix {
     virtual SetBitPositions get_row(Row row) const = 0;
     virtual std::vector<SetBitPositions> get_rows(const std::vector<Row> &rows) const;
     virtual std::vector<Row> get_column(Column column) const = 0;
+
     // get all selected rows appended with -1 and concatenated
     virtual std::vector<Column> slice_rows(const std::vector<Row> &rows) const;
 
+    virtual void slice_columns(const std::vector<Column> &columns,
+                               const ColumnCallback &callback) const;
```

Comment:
```suggestion
    virtual void slice_columns(const std::vector<Column> &columns,
                               const std::function<void(Column, bitmap&&)> &callback) const;
```

Reply:
I assume this should be parallelized? (otherwise why would you call the indexes `Column`).

So, pass num_threads as another argument too?